x/leverage/keeper/store.go

```diff
@@ -38,13 +38,16 @@ func (k Keeper) getAdjustedTotalBorrowed(ctx sdk.Context, denom string) sdk.Dec
 }
 
 // setAdjustedBorrow sets the adjusted amount borrowed by an address in a given denom directly instead
-// of computing it from real borrowed amount. Should only be used by genesis and SetBorrow, as other
-// functions deal in non-adjusted amounts using SetBorrow. Also updates AdjustedTotalBorrowed by the
+// of computing it from real borrowed amount. Should only be used by genesis and setBorrow, as other
+// functions deal in non-adjusted amounts using setBorrow. Also updates AdjustedTotalBorrowed by the
 // resulting changes in borrowed amount. If either amount to store is zero, any stored value is cleared.
 func (k Keeper) setAdjustedBorrow(ctx sdk.Context, addr sdk.AccAddress, borrow sdk.DecCoin) error {
 	if err := borrow.Validate(); err != nil {
 		return err
 	}
+	if addr.Empty() {
+		return types.ErrEmptyAddress
+	}
```

Comment:
Empty addresses are especially dangerous when interacting with store keys of the structure `prefix | lengthPrefixed(addr) | denom | 0x00` because the function `address.MustLengthPrefix(addr)` will return `[]byte{}` on empty input.

That would cause our getters and iterators to interpret the first byte of `denom` as the length prefix to an address, likely reaching out of slice bounds for the address, or interpreting part of a long IBC denom as an address, and the remainder as its own unique denom, causing all kinds of unpredictable behavior.

Since `addr.Empty` is computationally cheap I put it in all `x/leverage` functions with `store.Set` that take address as an input, even though such bad input should already be stopped before reaching such code.